python/lsst/analysis/tools/actions/plot/barPlots.py

```diff
@@ -0,0 +1,360 @@
+from lsst.pex.config import Config, ConfigDictField, DictField, Field
+from matplotlib.figure import Figure
+from matplotlib.gridspec import GridSpec
+from matplotlib.patches import Rectangle
+
+
+class BarPanel(Config):
+    label = Field[str](
+        doc="Panel x-axis label.",
+        default="label",
+    )
+    bars = DictField[str, str](
+        doc="A dict specifying the bar graphs to be plotted in this panel. Keys are used to identify "
+        "bar graph IDs. Values are used to add to the legend label displayed in the upper corner of the "
+        "panel.",
+        optional=False,
+    )
+    yscale = Field[str](
+        doc="Y axis scaling.",
+        default="linear",
+    )
+
+
+class BarPlot(PlotAction):
+    panels = ConfigDictField(
+        doc="A configurable dict describing the panels to be plotted, and the bar graphs for each panel.",
+        keytype=str,
+        itemtype=BarPanel,
+        default={},
+    )
+    cmap = Field[str](
+        doc="Color map used for bar lines. All types available via `plt.cm` may be used. "
+        "A number of custom color maps are also defined: `newtab10`, `bright`, `vibrant`.",
+        default="newtab10",
+    )
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        for panel in self.panels:  # type: ignore
+            for barData in self.panels[panel].bars.items():  # type: ignore
+                yield barData, Vector
+
+    def __call__(self, data: KeyedData, **kwargs) -> Mapping[str, Figure] | Figure:
+        return self.makePlot(data, **kwargs)
+
+    def makePlot(
+        self, data: KeyedData, plotInfo: Mapping[str, str] = None, **kwargs  # type: ignore
+    ) -> Figure:
+        """Make an N-panel plot with a user-configurable number of bar graphs
+        displayed in each panel.
+
+
+        Parameters
+        ----------
+        data : `KeyedData`
+            The catalog to plot the points from.
+        plotInfo : `dict`
+            An optional dictionary of information about the data being
+            plotted with keys:
+                `"run"`
+                    Output run for the plots (`str`).
+                `"tractTableType"`
+                    Table from which results are taken (`str`).
+                `"plotName"`
+                    Output plot name (`str`)
+                `"SN"`
+                    The global signal-to-noise data threshold (`float`)
+                `"skymap"`
+                    The type of skymap used for the data (`str`).
+                `"tract"`
+                    The tract that the data comes from (`int`).
+                `"bands"`
+                    The bands used for this data (`str` or `list`).
+                `"visit"`
+                    The visit that the data comes from (`int`)
+
+        Returns
+        -------
+        fig : `matplotlib.figure.Figure`
+            The resulting figure.
+
+        """
+
+        # set up figure
+        fig = plt.figure(dpi=800)
```

Comment:
Do we need this high a dpi? Do the figures come out very large?

Reply:
It probably doesn't. I increased the dpi to test something Meredith was asking about and never changed it back. I'll lower it.